File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n = s.size(), ansIdx = -1;
        char ans = ' ';
        unordered_map<char,int>m;
        for(int i = 0; i < n; i++){
            ++m[s[i]];
        }
        for(int i = 0; i < n; i++){
            if(m[s[i]] == 1){
                return s.find(s[i]);
            }
        }
        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        
        int freq[] = new int[26];
        
        for(int i = 0; i < s.length(); i++)
            ++freq[s.charAt(i) - 'a'];
        
        int i = 0;
        
        for(; i < s.length(); i++)
            if(freq[s.charAt(i) - 'a'] == 1)   break;
        
        return (i == s.length()) ? -1 : i;
        
    }
}